.github/workflows/tests.yml

```diff
@@ -62,10 +62,18 @@ jobs:
       - name: Run unit tests
         run: make unit-tests
 
-      # disabled as 'pinging codecov' hangs
-      # see https://github.com/codecov/feedback/issues/354
-      # - name: Send coverage to codecov.io
-      #   run: bash <(curl -s https://codecov.io/bash)
+  codecov:
+    name: Run tests and collect coverage
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@v2
+       with:
+         fetch-depth: 2
```

Comment:
I believe these all are useless, IMO we do not need separate job for codecov.
I think we can just add it to existing unit test run. Or have you had a reson to split it into a separate job?